monkey/tests/monkey_island/cc/services/test_post_breach_files.py

```diff
@@ -0,0 +1,60 @@
+import os
+
+import pytest
+
+from monkey_island.cc.services.post_breach_files import PostBreachFilesService
+
+
+def raise_(ex):
+    raise ex
+
+
+@pytest.fixture(autouse=True)
+def custom_pba_directory(tmpdir):
+    PostBreachFilesService.initialize(tmpdir)
+
+
+def create_custom_pba_file(filename):
+    assert os.path.isdir(PostBreachFilesService.get_custom_pba_directory())
+
+    file_path = os.path.join(PostBreachFilesService.get_custom_pba_directory(), filename)
+    open(file_path, "a").close()
+
+
+def test_remove_pba_files():
+    create_custom_pba_file("linux_file")
+    create_custom_pba_file("windows_file")
+
+    custom_pda_dir_contents = os.listdir(PostBreachFilesService.get_custom_pba_directory())
+    assert len(custom_pda_dir_contents) == 2
+
+    PostBreachFilesService.remove_PBA_files()
+
+    custom_pda_dir_contents = os.listdir(PostBreachFilesService.get_custom_pba_directory())
+    assert len(custom_pda_dir_contents) == 0
+
+
+@pytest.mark.skipif(os.name != "posix", reason="Tests Posix (not Windows) permissions.")
+def test_custom_pba_dir_permissions():
+    st = os.stat(PostBreachFilesService.get_custom_pba_directory())
+
+    assert st.st_mode == 0o40700
+
+
+def test_remove_failure(monkeypatch):
+    monkeypatch.setattr(os, "remove", lambda x: raise_(OSError("Permission denied")))
```

Comment:
Why not just `raise(OSError("Permission denied"))`? This would avoid the need for raise_ which does the same thing.

Reply:
Trying to raise an exception in a lamdba function results in a syntax error. https://stackoverflow.com/questions/8294618/define-a-lambda-expression-that-raises-an-exception

Reply:
I suspect this raiser would be useful in other tests as well, maybe let's move it to some common/utils file?